arch/src/x86_64/layout.rs

```diff
@@ -107,6 +107,11 @@ pub const KVM_TSS_SIZE: u64 = (3 * 4) << 10;
 pub const KVM_IDENTITY_MAP_START: GuestAddress = GuestAddress(KVM_TSS_START.0 + KVM_TSS_SIZE);
 pub const KVM_IDENTITY_MAP_SIZE: u64 = 4 << 10;
 
+/// TPM Address Space Range
+/// This Address rangs is specific to CRB Interface
+pub const TPM_START: GuestAddress = GuestAddress(0xfed4_0000);
```

Comment:
So you're saying the spec reserves this address? That's weird for a specification to put such constraint on the implementation, especially given it might not fit every architecture.
I thought the point of TPM2 ACPI table was to advertise where to find the TPM, meaning we wouldn't have to hardcode this value.

Reply:
Ok well I guess there's no way around it. That being said, please make sure you reserve this specific address range in the allocator so that the allocator won't allow other devices to be positioned on this range.

Reply:
will handle this in next revision.

Reply:
@praveen-pk That is the function for reserving PIO addresses not MMIO addresses.

Reply:
@rbradford, should I use the following mmio allocator instead?
```
        self.address_manager.allocator
            .lock()
            .unwrap()
            .allocate_platform_mmio_addresses(...)
```

Reply:
I don't know if your address is covered by an allocator so you might not need to do anything but `.allocate_io_addresses()` is definitely wrong.

Reply:
@praveen-pk you should use the MMIO allocator. And I think this is needed so that we don't end up with the allocator thinking this is a free address range.

Reply:
@sboeuf unfortunately, using `allocate_platform_mmio_addresses()` to reserve TPM address range doesn't work.

In my test configuration `platform_mmio_address` range is configured at: `0x3fff_ffef_0000 - 0x3fff_fffe_ffff`. This range is computed at https://github.com/cloud-hypervisor/cloud-hypervisor/blob/6722c303b05c0b3b7f751ec43927db315d4cd52c/vmm/src/memory_manager.rs#L889.

As the TPM address range doesn't fall within the above range, https://github.com/cloud-hypervisor/cloud-hypervisor/blob/6722c303b05c0b3b7f751ec43927db315d4cd52c/vm-allocator/src/address.rs#L93 check fails and cloud-hypervisor exits with error.

```
Error booting VM: VmBoot(DeviceManager(AllocateMmioAddress))
```

MMIO Hole range is `0xC000_0000 - 0xE800_0000`. TPM address range doesn't fall in this range either.

Reply:
I see, then you don't need to add it to any allocator as there's some kind of grey zone from `0xe800000` to `0x100000000`. I think looking at the IOAPIC is a good example. Given it's located at `0xfec00000`, there's no need to register it to any allocator.

Reply:
Gotcha, dropped the allocator step.